pilot/pkg/networking/plugin/authz/rbac_test.go

```diff
@@ -1027,9 +1201,163 @@ func TestConvertRbacRulesToFilterConfig(t *testing.T) {
-	policy12 := generateSimplePolicyForNotRuleWithHeader(methodHeader, "DELETE")
-	policy13 := generateSimplePolicyForNotRuleWithHeader(pathHeader, "/secret_path")
+	policy11 := &policy.Policy{
+		Permissions: []*policy.Permission{
+			{
+				Rule: &policy.Permission_AndRules{
+					AndRules: &policy.Permission_Set{
+						Rules: []*policy.Permission{
+							{
+								Rule: &policy.Permission_NotRule{
+									NotRule: &policy.Permission{
+										Rule: generateHeaderRule([]*route.HeaderMatcher{
+											{Name: hostHeader, HeaderMatchSpecifier: &route.HeaderMatcher_ExactMatch{ExactMatch: "finances.google.com"}},
+										}),
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		Principals: []*policy.Principal{{
+			Identifier: &policy.Principal_AndIds{
+				AndIds: &policy.Principal_Set{
+					Ids: []*policy.Principal{
+						{
+							Identifier: &policy.Principal_OrIds{
+								OrIds: &policy.Principal_Set{
+									Ids: []*policy.Principal{
+										{
+											Identifier: &policy.Principal_Metadata{
+												Metadata: generateMetadataListMatcher(authn.AuthnFilterName,
+													[]string{attrRequestClaims, "groups"}, "*-group"),
+											},
+										},
+									},
+								},
+							},
+						},
+						{
+							Identifier: &policy.Principal_NotId{
+								NotId: &policy.Principal{
+									Identifier: &policy.Principal_OrIds{
+										OrIds: &policy.Principal_Set{
+											Ids: []*policy.Principal{
+												{
+													Identifier: &policy.Principal_Metadata{
+														Metadata: generateMetadataListMatcher(authn.AuthnFilterName,
+															[]string{attrRequestClaims, "groups"}, "deprecated-group"),
+													},
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		}},
+	}
+
+	policy12 := &policy.Policy{
+		Permissions: []*policy.Permission{
+			{
+				Rule: &policy.Permission_AndRules{
+					AndRules: &policy.Permission_Set{
+						Rules: []*policy.Permission{
+							{
+								Rule: &policy.Permission_NotRule{
+									NotRule: &policy.Permission{
+										Rule: generateHeaderRule([]*route.HeaderMatcher{
+											{Name: methodHeader, HeaderMatchSpecifier: &route.HeaderMatcher_ExactMatch{ExactMatch: "DELETE"}},
+										}),
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		Principals: []*policy.Principal{{
+			Identifier: &policy.Principal_AndIds{
+				AndIds: &policy.Principal_Set{
+					Ids: []*policy.Principal{
+						{
+							Identifier: &policy.Principal_OrIds{
+								OrIds: &policy.Principal_Set{
+									Ids: []*policy.Principal{
+										{
+											Identifier: &policy.Principal_Metadata{
+												Metadata: generateMetadataStringMatcher(
+													attrSrcPrincipal, &metadata.StringMatcher{
+														MatchPattern: &metadata.StringMatcher_Regex{
+															Regex: ".*",
```

Comment:
Let me comment here to make my previous question on `allAuthenticatedUsers` more clear. It looks like the `allAuthenticatedUsers` is generated as a metadata string matcher with regular matching ".*", could you confirm does this only match on authenticated user and doesn't match on un-authenticated users?

Reply:
Thanks for looking into this @yangminzhu 
Actually I was mistaken before. It does not generate Principal_Authenticated. In fact, after looking at the code, only TCP filter uses Principal_Authenticated. It seems like this was done like this long before. 
When `allAuthenticatedUsers` is used, it acts exactly as `source.principal` in RBAC, using filter: "istio_authn". Below is the envoy config.

```
                        {
                         "metadata": {
                          "filter": "istio_authn",
                          "path": [
                           {
                            "key": "source.principal"
                           }
                          ],
                          "value": {
                           "string_match": {
                            "regex": ".*"
                           }
                          }
                         }
                        }
```

Reply:
We do this because we don't have authn filter for TCP which means we don't have those metadata attributes for TCP, so we use `Principal_Authenticated` instead.

Back to my original question, we need to make sure the `istio_authn` metadata matcher only matches on authenticated users. You have the regular expression `.*` in the matcher which generally means it matches to any string, do we know does it also match to unauthenticated user? (I feel not but we need to check to confirm otherwise it will be a bug)

The second thing is we need to make sure `allAuthenticatedUsers` is supported in both HTTP and TCP, which means when in TCP case, we need to use  `Principal_Authenticated` instead.

Reply:
From the implementation perspective, setting `names = ["allAuthenticatedUsers"]` is the same as the old `source.principal = "*"`. I'm having some problem with my cluster and bookinfo example, so I cannot manually test it now :( But if `source.principal = "*"` works (for TCP and `istio_authn`), then `allAuthenticatedUsers` should work.

Reply:
That makes sense, so basically we have the answer "Yes" here.

Reply:
Great. I'll mark this as resolved now. Feel free to re-open it.